homeassistant/components/starlink/coordinator.py

```diff
@@ -58,14 +58,14 @@ def __init__(self, hass: HomeAssistant, name: str, url: str) -> None:
     async def _async_update_data(self) -> StarlinkData:
         async with asyncio.timeout(4):
             try:
-                status, location, sleep = await asyncio.gather(
-                    self.hass.async_add_executor_job(status_data, self.channel_context),
-                    self.hass.async_add_executor_job(
-                        location_data, self.channel_context
-                    ),
-                    self.hass.async_add_executor_job(
-                        get_sleep_config, self.channel_context
-                    ),
+                status = await self.hass.async_add_executor_job(
+                    status_data, self.channel_context
+                )
+                location = await self.hass.async_add_executor_job(
+                    location_data, self.channel_context
+                )
+                sleep = await self.hass.async_add_executor_job(
```

Comment:
Instead of scheduling three executor jobs sequentially, we should rewrite this to just schedule one executor job and do all the sync work in a single function. Create a function that calls `status_data`, `location_data` and `get_sleep_config` and schedule the function once on the executor.

Reply:
I commented this on the previous PR two weeks ago:
https://github.com/home-assistant/core/pull/103057#discussion_r1529277560